File: bitget-node-sdk-api/src/lib/v1/SpotAccountApi.ts
import {BaseApi} from '../BaseApi';

export class SpotAccountApi extends BaseApi {

    getInfo() {
        const url = '/api/spot/v1/account/getInfo';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    assetsLite(qsOrBody: object) {
        const url = '/api/spot/v1/account/assets-lite';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    bills(qsOrBody: object) {
        const url = '/api/spot/v1/account/bills';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    transferRecords(qsOrBody: object) {
        const url = '/api/spot/v1/account/transferRecords';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

}

File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v1/SpotAccountApi.java
package com.bitget.openapi.api.v1;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface SpotAccountApi {

    @GET("/api/spot/v1/account/getInfo")
    Call<ResponseResult> getInfo(@QueryMap Map<String, String> paramMap);

    @GET("/api/spot/v1/account/assets-lite")
    Call<ResponseResult> assetsLite(@QueryMap Map<String, String> paramMap);

    @POST("/api/spot/v1/account/bills")
    Call<ResponseResult> bills(@Body Map<String, String> paramMap);

    @GET("/api/spot/v1/account/transferRecords")
    Call<ResponseResult> transferRecords(@QueryMap Map<String, String> paramMap);
}
